Update to latest Electron fixing security issues

mountainash opened the pull request:
Changing from the current master v1.6.11 of Electron to v9.1.0

mountainash committed:
electron-updater was so far behind it wouldn't build

mountainash committed:
Updated dependencies

 - added meta data to the `package.json`

mountainash committed:
make work with older version of Electron

Downgraded Electon to 3.1.13 probably as webview is to strict these days (for a reason)

mountainash committed:
updated from depreicated makeSingleInstance

 - initial window wider

mountainash committed:
Hiding broken App menu items

 - until they can be fixed again

mountainash committed:
FIX: getting os.release for Report Issue menu

mountainash committed:
updated and compressed images

- new screenshot

mountainash committed:
Added session persistency - no need to login again each launch

- removed restrictive window settings

Closes https://github.com/kreativgebiet/taskana/issues/5

mountainash committed:
New Task key combo working again

- Disabled New Section

Related to: https://github.com/kreativgebiet/taskana/issues/6

mountainash committed:
Removed old CSS (doen't map anymore)

- hiding some over sell
- Fixed: https://github.com/kreativgebiet/taskana/issues/4

mountainash committed:
Disabled "View" menu

- until hooked back up again

Releated: https://github.com/kreativgebiet/taskana/issues/6

mountainash committed:
Remembering window position/size between use

- removed 'electron-store'
- added 'electron-config'
- commenting out all the unused "vibrancy" stuff

renovate-bot committed:
Add renovate.json

mountainash committed:
"Services" menu enabled

- hiding new upgrade button
- removed yarn.lock

mountainash committed:
Merge pull request #1 from mountainash/renovate/configure

Configure Renovate

renovate-bot committed:
Pin dependencies

renovate-bot committed:
Update dependency electron to v9

mountainash committed:
Merge pull request #2 from mountainash/renovate/pin-dependencies

Pin dependencies

mountainash committed:
Updated to latest dependencies

mountainash committed:
Merge branch 'renovate/electron-9.x'

renovate-bot committed:
Pin dependencies

mountainash committed:
Merge pull request #12 from mountainash/renovate/pin-dependencies

Pin dependencies

iduuck reviewed (commented):
First of all, thanks for your contribution. Really stoked since this is a super old project of us, bringing it back to live again.

Secondly, I find a little confusing that everything is just commented out. Please remove the files you don't want to use anymore.

iduuck commented:
On another note, what I found to pull Taskana out of it's competitors is, that there is no actual "title bar", so I would like to see this, too.

mountainash commented:
> Secondly, I find a little confusing that everything is just commented out. Please remove the files you don't want to use anymore.

I left it there like a TODO incase someone want to get it working again - but I'm more than happy to delete it all.

mountainash commented:
> what I found to pull Taskana out of it's competitors is, that there is no actual "title bar", so I would like to see this, too.

I don't think it would be that usable without the title bar with the current design of Asana. It might have worked well when Asana looked like this:
![Taskana Tasks sample image](https://raw.githubusercontent.com/kreativgebiet/taskana/master/build/screenshot.png)

But I can't see it looking/working better for the current Asana design: 

![Taskana Tasks sample image](https://raw.githubusercontent.com/mountainash/taskana/develop/build/screenshot.png)

mountainash committed:
Removing unused code

As requested in https://github.com/kreativgebiet/taskana/pull/9#pullrequestreview-450767798

mountainash commented:
I've created a new Issue for the title bar "debate". #10

renovate-bot committed:
Update dependency electron to v9.3.1

mountainash committed:
Version number bump

Tested on electron 9.3.1

mountainash committed:
Merge pull request #13 from mountainash/renovate/electron-9.x

Update dependency electron to v9.3.1

mountainash committed:
kreativgebiet has been archived - updated Gtihub project links

Updated Asana short urls to link direct to HTTPS

Closes #21

mountainash committed:
Merge pull request #23 from mountainash/bye-kreativgebiet

kreativgebiet has been archived - updated Gtihub project links

mountainash committed:
Updating Renovate Bot PR's in one commit

Updated to Electron v10.1.4
Updated to electron-packager v15.1.0
Updated to electron-log v4.2.4
Updated to electron-updater v4.3.5
Updated to electron-builder v22.9.1

Closes #19

mountainash committed:
Merge pull request #25 from mountainash/dependency-update

Updating Renovate Bot PR's in one commit

mountainash committed:
npm audit fix (dot-prop v4.2.1)

found 1 high severity vulnerability

renovate-bot committed:
Update dependency electron to v10.1.5

mountainash committed:
Merge pull request #26 from mountainash/renovate/electron-10.x

Update dependency electron to v10.1.5

mountainash committed:
Added `publish` script

mountainash committed:
Changed "publish" script to "release"

mountainash committed:
Changing publishing/release location

renovate-bot committed:
Update dependency electron-log to v4.3.0

mountainash committed:
Merge pull request #28 from mountainash/renovate/electron-log-4.x

Update dependency electron-log to v4.3.0

mountainash committed:
Added context menu support

renovate-bot committed:
Pin dependency electron-context-menu to 2.3.0

renovate-bot committed:
Update dependency electron-packager to v15.2.0

renovate-bot committed:
Update dependency electron to v10.2.0

renovate-bot committed:
Update dependency electron-log to v4.3.1

renovate-bot committed:
Update dependency electron-debug to v3.2.0

renovate-bot committed:
Update dependency electron to v11

mountainash committed:
Merge pull request #29 from mountainash/renovate/pin-dependencies

Pin dependency electron-context-menu to 2.3.0

renovate-bot committed:
Update dependency electron-context-menu to v2.4.0

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-debug-3.x' into 2021-jan-dependency-updates

mountainash committed:
Merge branch 'renovate/electron-log-4.x' into 2021-jan-dependency-updates

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-packager-15.x' into 2021-jan-dependency-updates

mountainash committed:
Bump ini from 1.3.5 to 1.3.8

Closes #33

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-10.x' into 2021-jan-dependency-updates

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-11.x' into 2021-jan-dependency-updates

mountainash committed:
Version bump :point_up:

mountainash committed:
Merge pull request #37 from mountainash/2021-jan-dependency-updates

2021 jan dependency updates

mountainash committed:
`npm update`

mountainash committed:
Renovate dep updates - Electron v12

Closes #40 #41 #42 #43 #44 & #45

renovate-bot committed:
Update dependency electron to v12.0.2

mountainash committed:
Merge pull request #45 from mountainash/renovate/electron-12.x

Update dependency electron to v12.0.2

mountainash committed:
Added website to readme

mountainash committed:
Dependency updates

Closes #46 #47 #49 #50

mountainash committed:
Only calling `log` dep when needed

mountainash committed:
Saving login details in the KeyStore

closes #51

mountainash committed:
Merge pull request #52 from mountainash/51-loginkeychain

Store login

mountainash committed:
Tidy up

mountainash committed:
Triggering `change` event on login inputs to get value into virtual DOM

mountainash committed:
Safer handling of `openExternal`

Help from: https://gist.github.com/MarshallOfSound/ad19a8846c06033cb07a8b203ebda7f6

mountainash committed:
Merge branch 'master' into 51-loginkeychain

mountainash committed:
Merge pull request #53 from mountainash/51-loginkeychain

51 loginkeychain + Safer `openExternal` handling

mountainash committed:
Dependency updates

Closes #47 #50 #54 #55 #56

mountainash committed:
Allowing Electron to handle Asana's AWS file server

Electron (on macOS) handles the download nicely with a nicer/simpler dialogue than the browsers.

Closes #57

mountainash committed:
New Menu Items

- added "Preferences"
- Added "Keyboard Shortcuts" menu item to Asana documentation
- Also, README updated with info about Asana official app

mountainash committed:
Changed Preferences short cut key from `;` to `,`

mountainash committed:
Merge pull request #58 from mountainash/develop

New Menu items + Better download handeling

renovate-bot committed:
Update dependency electron to v13.1.1

mountainash committed:
Merge pull request #56 from mountainash/renovate/electron-13.x

Update dependency electron to v13.1.1

mountainash committed:
Version bump for Electron. Removed unused `electron-packager` from deps

mountainash committed:
Updated dependencies fixes external site browser launch freeze

- Update dependency `electron` from v13.1.1 to **v3.1.7**
- Update dependency `electron-builder` from v22.11.5 to **v22.11.7**
- Update dependency `electron-context-menu` from v3.1.0 to **v3.1.1**
- Update dependency `electron-log` from v4.3.5 to **v4.4.0**

renovate-bot committed:
Update dependency electron-log to v4.4.1

renovate-bot committed:
Update dependency electron to v13.5.1

renovate-bot committed:
Update dependency electron to v15

mountainash committed:
Patch version bump

mountainash committed:
Removed dev deps and allowing .doubleclick.net

mountainash committed:
Merge pull request #66 from mountainash/develop

Removing Dev Deps

mountainash committed:
Deps Update: Electron patch

renovate-bot committed:
Update dependency electron-builder to v22.13.1

Shylajhaa committed:
Merge remote-tracking branch 'upstream/master' into cmd-n-fix

Shylajhaa committed:
Fixed issue with command + N not opening a popup

mountainash committed:
Merge pull request #68 from Shylajhaa/cmd-n-fix

Fixed issue with command + N not opening a popup

Shylajhaa committed:
Merge remote-tracking branch 'upstream/master' into pref-fix

Shylajhaa committed:
Fixed issue in preferences tab opening wrong link

Position of settings option changed in menu list

mountainash committed:
Merge pull request #69 from Shylajhaa/pref-fix

Fixed issue in preferences tab opening wrong link

mountainash committed:
Merge branch 'master' into develop

mountainash committed:
Updated Electron version and App version ready for a new build

mountainash committed:
Merge pull request #67 from mountainash/renovate/electron-builder-22.x

Update dependency electron-builder to v22.13.1

mountainash committed:
Dependency Updates, and Apple Silicon build target added

mountainash committed:
Merge branch 'master' into develop

mountainash committed:
Dependency Updates

mountainash committed:
Merge pull request #75 from mountainash/develop

Dependency Updates

renovate-bot committed:
Update dependency electron-updater to v4.6.5

renovate-bot committed:
Update dependency keytar to v7.9.0

renovate[bot] committed:
Update dependency electron-store to v8.0.2

renovate[bot] committed:
Update dependency electron-context-menu to v3.3.0

renovate[bot] committed:
Update dependency electron to v19

renovate[bot] committed:
Update dependency electron-builder to v23

mountainash committed:
Comment fix

mountainash committed:
Merge branch 'develop' into renovate/electron-builder-23.x

mountainash committed:
Merge branch 'develop' into renovate/electron-19.x

mountainash committed:
Merge branch 'develop' into renovate/electron-store-8.x

mountainash committed:
Merge branch 'develop' into renovate/keytar-7.x

mountainash committed:
Merge branch 'develop' into renovate/electron-context-menu-3.x

mountainash committed:
Merge branch 'renovate/electron-19.x' into develop

mountainash committed:
Merge branch 'renovate/keytar-7.x' into develop

mountainash committed:
Merge branch 'renovate/electron-builder-23.x' into develop

mountainash committed:
Merge branch 'renovate/electron-store-8.x' into develop

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-updater-4.x' into develop

mountainash committed:
Fix for Command+N (Asana has a new menu)

mountainash committed:
Accelerator changes

As per https://www.electronjs.org/docs/latest/api/accelerator#platform-notice

mountainash committed:
Merge pull request #93 from mountainash/develop

Dep. Updates

mountainash committed:
Dep updates

mountainash committed:
Merge pull request #94 from mountainash/develop

Dep updates

mountainash committed:
Adding github sponsor username

mountainash committed:
Adding github sponsor username

mountainash committed:
Adding VScode workspace colouring

mountainash committed:
Electron update

Closes #100

mountainash committed:
Electron Builder update

Closes #96

mountainash committed:
electron-updater updated

Closes #97

mountainash committed:
Update electron-context-menu

Closes #99

mountainash committed:
Update electron-store

Closes #95

mountainash committed:
Merge branch 'develop' of https://github.com/mountainash/taskana into develop

mountainash committed:
JS error fox to allow key to work

mountainash committed:
Removing unused images

mountainash committed:
Merge pull request #101 from mountainash/develop

Big bunch of dependency updates

mountainash committed:
🚸 Fixed: Shorts-cuts + deps updates

mountainash committed:
Bump `electron-builder` to v24.2.1 to close #115

mountainash committed:
Merge pull request #116 from mountainash/develop

v1.16.1

mountainash committed:
Filesize "selling" point

mountainash committed:
📦 Dependency updates

mountainash committed:
Selector updates to match current DOM on asana.com

mountainash committed:
Merge pull request #125 from mountainash/develop

v1.16.0

renovate[bot] committed:
Update dependency electron-store to v10

renovate[bot] committed:
Update dependency electron-context-menu to v4

renovate[bot] committed:
Update dependency electron to v32

renovate[bot] committed:
Update dependency electron-updater to v6.3.0 [SECURITY]

renovate[bot] committed:
Update dependency electron-builder to v25

mountainash committed:
Merge remote-tracking branch 'origin/renovate/npm-electron-updater-vulnerability' into mountainash/issue61

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-builder-25.x' into mountainash/issue61

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-store-10.x' into mountainash/issue61

mountainash committed:
Merge remote-tracking branch 'origin/renovate/electron-context-menu-4.x' into mountainash/issue61

mountainash committed:
:page_facing_up: README linting

mountainash committed:
:package: electron-updater v6.3.0 to v6.3.9

mountainash committed:
:construction_worker: updated from CommonJS requires to ESM imports

mountainash committed:
:hammer: fix to allow `browser.js` to load into the website

mountainash committed:
Merge pull request #141 from mountainash/mountainash/issue61

📦  Updates to all Deps. Changes from CommonJS to ESM input/exports
